packages/components/src/pages/VsCodeExtension.vue

```diff
@@ -401,16 +401,10 @@ export default {
       currentTraceFilterIndex: 0,
       showShareModal: false,
       shareURL: undefined,
+      shareEnabled: false,
```

Comment:
What's the benefit of moving this to a data property?

Reply:
I moved it to a data property because I wanted to use a feature flag. In VS Code, we use the `ExtensionSettings` class to keep track of feature flags, and it seemed easier to import that class into the `appmapEditorProvider.ts` file rather than `appmapView.js` (see VS Code PR [here](https://github.com/getappmap/vscode-appland/pull/608/commits/1fdf70b4943a56cfc9d582e3dc686e47e702c96e)).

Reply:
You're right, importing backend logic into the frontend won't work because the VS Code API isn't available within the context of the browser. Instead we need a channel of communication between the backend and frontend to exchange information like feature flags.

We're doing this to a greater extent in the instructions web view. Here, we take the initial state (which is probably enough in this case) and initialize the root Vue applications data properties:
https://github.com/getappmap/vscode-appland/blob/master/web/src/installGuideView.js#L34

Then the flag is provided via props:
https://github.com/getappmap/vscode-appland/blob/master/web/src/installGuideView.js#L24

The only reason we're using a data property here is because we want the flag to be reactive. By modifying the data property later on, it's propagated to any child elements where it's passed as a prop. We have another message handler to handle the state change explicitly:
https://github.com/getappmap/vscode-appland/blob/master/web/src/installGuideView.js#L109-L114

So that's all to say that I don't think we need this change. We just need to supply the feature flag boolean to the prop from within the extension frontend. After all, there's minimal difference between passing a value via a prop or via method call on creation. We're just setting a flag, so there shouldn't be any technical limitation we're hitting.